### PR TITLE
Adding Support for alert channel

### PR DESCRIPTION
**Pull request by Bhavneet-Sharma**
# Description
Adding Support for alert channels

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [x] I have performed Ansible Sanity test using --docker default
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit Test
- [x] Functional Testing

### CODE REVIEW

File: plugins/modules/alert_channel.py [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+                    "system": False,
+                    "type": channel_params['type']
+                }
+                return diff_dict
+            else:
+                diff_dict = copy.deepcopy(channel_details)
+                modify_dict = self.is_alert_channel_modify_required(channel_params, diff_dict)
+                for key in modify_dict.keys():
+                    diff_dict[key] = modify_dict[key]
+                return diff_dict
```
Comment (Kritika-Bhateja-03) [inline plugins/modules/alert_channel.py:510]: As at line 504 , we return the same diff_dict , lets keep return statement outside of if else statements.

Comment (Bhavneet-Sharma) [reply plugins/modules/alert_channel.py:510]: Fixed.

**Review by felixs88 — DISMISSED** [review]

**Review by Kritika-Bhateja-03 — DISMISSED** [review]

**Review by felixs88 — DISMISSED** [review]

File: playbooks/modules/alert_channel.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  connection: local
+  vars:
+    onefs_host: "x.x.x.x"
+    api_user: "user"
+    api_password: "xxxx"
+    verify_ssl: false
+    channel_smtp: "sample_event_channel"
+    channel_connectemc: "connect_emc_alert_channel"
+
+  tasks:
```
Comment (trisha-dell) [inline playbooks/modules/alert_channel.yml:13]: please add a task to send test alert.

Comment (Bhavneet-Sharma) [reply playbooks/modules/alert_channel.yml:13]: Done.

**Review by trisha-dell — APPROVED** [review]

**Review by sachin-apa — DISMISSED** [review]

**Review by Kritika-Bhateja-03 — APPROVED** [review]

**Review by trisha-dell — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**